Description: avoid feature dist-server
 Some needed crates are not yet available in Debian.
Author: Jonas Smedegaard <dr@jones.dk>
Forwarded: not-needed
Last-Update: 2023-08-24
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -47,7 +47,6 @@
 hyper = { version = "0.14.27", optional = true, features = ["server"] }
 is-terminal = "0.4.12"
 jobserver = "0.1"
-jwt = { package = "jsonwebtoken", version = "9", optional = true }
 libc = "0.2.153"
 linked-hash-map = "0.5"
 log = "0.4"
@@ -59,7 +58,6 @@
 object = "0.32"
 once_cell = "1.19"
 opendal = { version = "0.48.0", optional = true, default-features = false }
-openssl = { version = "0.10.64", optional = true }
 rand = "0.8.4"
 regex = "1.10.3"
 reqsign = { version = "0.16.0", optional = true }
@@ -99,20 +97,6 @@
 zip = { version = "0.6", default-features = false }
 zstd = "0.13"
 
-# dist-server only
-nix = { version = "0.28.0", optional = true, features = [
-  "mount",
-  "user",
-  "sched",
-  "signal",
-  "process",
-] }
-rouille = { version = "3.6", optional = true, default-features = false, features = [
-  "ssl",
-] }
-syslog = { version = "6", optional = true }
-version-compare = { version = "0.1.1", optional = true }
-
 [dev-dependencies]
 assert_cmd = "2.0.12"
 cc = "1.0"
@@ -126,10 +110,6 @@
 [target.'cfg(unix)'.dependencies]
 daemonize = "0.5"
 
-[target.'cfg(not(target_os = "freebsd"))'.dependencies.libmount]
-optional = true
-version = "0.1.15"
-
 [features]
 all = [
   "dist-client",
@@ -155,7 +135,7 @@
 # Enable features that will build a vendored version of openssl and
 # statically linked with it, instead of linking against the system-wide openssl
 # dynamically or statically.
-vendored-openssl = ["openssl?/vendored", "reqwest?/native-tls-vendored"]
+vendored-openssl = ["reqwest?/native-tls-vendored"]
 # Enable features that require unstable features of Nightly Rust.
 unstable = []
 # Enables distributed support in the sccache client
@@ -166,20 +146,8 @@
   "url",
   "sha2",
 ]
-# Enables the sccache-dist binary
-dist-server = [
-  "jwt",
-  "flate2",
-  "libmount",
-  "nix",
-  "openssl",
-  "reqwest",
-  "rouille",
-  "syslog",
-  "version-compare",
-]
 # Enables dist tests with external requirements
-dist-tests = ["dist-client", "dist-server"]
+dist-tests = ["dist-client"]
 
 [workspace]
 exclude = ["tests/test-crate"]
